### PR TITLE
Custom src and iface

### PR DESCRIPTION
**Pull request by oystub**
Required changes for https://github.com/aviant-tech/PX4-Autopilot/pull/36 in order to send messages with a custom source node id and and interface mask.

### CODE REVIEW

File: libuavcan/src/transport/uc_transfer_sender.cpp [hunk 1]
```diff
@@ -28,7 +28,50 @@ int TransferSender::send(const uint8_t* payload, unsigned payload_len, Monotonic
...
+    return send(payload, payload_len, tx_deadline, blocking_deadline, transfer_type,
+                dispatcher_.getNodeID(), dst_node_id, 0xff);
+}
+
+int TransferSender::send(const uint8_t* payload, unsigned payload_len, MonotonicTime tx_deadline,
+                         MonotonicTime blocking_deadline, TransferType transfer_type, NodeID src_node_id,
+                         NodeID dst_node_id, uint8_t iface_mask) const
+{
+    /*
+     * TODO: TID is not needed for anonymous transfers, this part of the code can be skipped?
```
Comment (Torjusba) [inline libuavcan/src/transport/uc_transfer_sender.cpp:47]: TODO

**Review by kibidev — CHANGES_REQUESTED** [review]
The code seem fine, but I have some comments regarding branches.
I do not think we should merge to main, as this makes syncing the upstream repository harder. Create a new "aviant/main" branch or similar to use instead.
Possible, we should also maintain something like "aviant/python3.10-fix" as long as this is the branched used in PX4.

It would be worth trying to get this merged in upstream, so we have less to maintain.

**Review by Torjusba — APPROVED** [review]
Looks good to me. Consider handling the TODO comment, but it does not seem like it is important to do anything about

File: libuavcan/src/transport/uc_transfer_sender.cpp (hunk 1 above)
Comment (oystub) [reply libuavcan/src/transport/uc_transfer_sender.cpp:47]: This todo comment was in the original code as well :joy:

**Pull request merged**